File: lib/csharp/UtaFormatix.Data/Note.cs
namespace UtaFormatix.Data;

/// <summary>Note model.</summary>
///
/// <param name="Key">Semitone value of the note's key (Center C = 60).</param>
/// <param name="TickOn">Tick position of the note's start.</param>
/// <param name="TickOff">Tick position of the note's end.</param>
/// <param name="Lyric">Lyric of the note.</param>
/// <param name="Phoneme">Phoneme of the note (if available).</param>
public record Note(
    int Key,
    long TickOn,
    long TickOff,
    string Lyric,
    string? Phoneme);

File: lib/kotlin-mpp/src/commonMain/kotlin/com/sdercolin/utaformatix/data/Note.kt
package com.sdercolin.utaformatix.data

import kotlinx.serialization.Serializable

/**
 * Note model
 * @param key Semitone value of the note's key (Center C = 60)
 * @param tickOn Tick position of the note's start
 * @param tickOff Tick position of the note's end
 * @param lyric Lyric of the note
 * @param phoneme Phoneme of the note (if available)
 */
@Serializable
data class Note(
    val key: Int,
    val tickOn: Long,
    val tickOff: Long,
    val lyric: String = "",
    val phoneme: String? = null,
)
